schema.json

```diff
@@ -1,255 +1,266 @@
-            }
-          },
-          "required": ["name", "type"]
+        "ranges": {
+          "description": "A list of possible ranges for the value. The
+                          complete range is the union of all the ranges and
+                          labels.",
+          "type": "array",
+          "items": { "$ref": "#/definitions/rangeType" },
+          "uniqueItems": true
         },
-        {
-          "properties": {
-            "type": {
-              "enum": ["group"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            },
-            "fields": {
-              "type": "array",
-              "items": {
-                "$ref": "#/definitions/field"
-              }
-            },
-            "max_size": {
-              "type": "integer",
-              "minimum": 0,
-              "exclusiveMinimum": true
-            },
-            "min_size": {
-              "type": "integer",
-              "minimum": 0
-            }
-          },
-          "required": ["name", "fields", "type"]
+        "unit": {
+          "description": "The unit type, defined in Table A-13 of E1.20.",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 255
         }
-      ]
+      },
+      "required": [ "type" ]
+    },
+    "labeledIntegerType": {
+      "description": "Associates a name to an integer value.",
+      "type": "object",
+      "properties": {
+        "name": { "$ref": "#/definitions/nameType" },
+        "value": { "type": "integer" }
+      },
+      "required": [ "value" ]
     },
-    "label": {
+    "listType": {
+      "description": "A list of objects all having the same type.",
       "type": "object",
       "properties": {
-        "label": {
-          "minLength": 1,
-          "type": "string"
+        "name": { "$ref": "#/definitions/nameType" },
+        "type": { "const": "list" },
+        "itemType": {
+          "$ref": "#/definitions/oneOfTypes"
         },
-        "value": {
+        "minItems": {
+          "description": "The minimum list size.",
+          "type": "integer",
+          "minimum": 0
+        },
+        "maxItems": {
+          "description": "The maximum list size.",
           "type": "integer",
-          "maximum": 4294967295,
           "minimum": 0
         }
       },
-      "required": ["label", "value"]
+      "required": [ "type", "itemType" ]
     },
-    "name": {
+    "nameType": {
+      "description": "A name is a string having a length of at least 1.",
       "type": "string",
       "minLength": 1
     },
-    "range": {
+    "oneOfTypes": {
+      "description": "One of any of the types. This provides a single location
+                      to keep the list.",
+      "oneOf": [
+        { "$ref": "#/definitions/bitFieldType" },
+        { "$ref": "#/definitions/compoundType" },
+        { "$ref": "#/definitions/integerType" },
+        { "$ref": "#/definitions/listType" },
```

Comment:
I think we need to drop listType otherwise you can have a list of lists and not know how to parse it (unless you enforce a fixed length on it), but then just doing a compoundType of e.g. three uint8's is easier and covers that edge case.

Reply:
The list type is needed for things such as a list of compound types, see DEFAULT_SLOT_VALUE, for example.